### PR TITLE
Refactoring for Workflow Scheduling and Bug Fixes

### PR DESCRIPTION
**Pull request by oumkale**
## Proposed changes

This PR has the Refactoring of workflow scheduling and bug fixes.

Custom Stepper Button Disabled for errors in Yaml
Reliability Score Button fix
Scheduling bug fixes
Renaming for Workflow name
Default value 10 for ReliabilityScore
Unrequired link are disabled Exam : VerifyCommit Github link
Yaml Editer Bug fixed (Full screen component)
CSS refactoring
Disabled edit option for scheduling time
Disabled Demo Launch

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [X] New feature (non-breaking change which adds functionality)
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

### CODE REVIEW

File: litmus-portal/frontend/src/components/FinishModal/index.tsx [hunk 1]
```diff
@@ -2,31 +2,27 @@ import Button from '@material-ui/core/Button';
...
-
-  return <img src="icons/finish.png" className={classes.mark} alt="mark" />;
-}
+import ButtonFilled from '../Button/ButtonFilled';
 
 const FinishModal = () => {
   const classes = useStyles();
 
   const [open, setOpen] = React.useState(false);
+  const [modified, setModified] = React.useState(false);
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/FinishModal/index.tsx:11]: What is the use of modified here?

Comment (oumkale) [reply litmus-portal/frontend/src/components/FinishModal/index.tsx:11]: To close the modal after clicking back to workflow.

File: litmus-portal/frontend/src/components/FinishModal/index.tsx [hunk 2]
```diff
@@ -37,30 +33,21 @@ const FinishModal = () => {
...
-        variant="contained"
-        color="primary"
-        data-cy="selectModal"
-        onClick={handleOpen}
-      >
-        Completed
-      </Button>
-
-      {/* Finish Modal is added */}
+      <div>
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/FinishModal/index.tsx:45]: Completed button not required, not in design?

File: litmus-portal/frontend/src/components/FinishModal/index.tsx (hunk 1 above)
Comment (Jonsy13) [reply litmus-portal/frontend/src/components/FinishModal/index.tsx:11]: We already have "open" for closing and opening the modal.

Comment (oumkale) [reply litmus-portal/frontend/src/components/FinishModal/index.tsx:11]: Thank you @Jonsy13 I resolved that

File: litmus-portal/frontend/src/components/FinishModal/index.tsx [hunk 3]
```diff
@@ -37,30 +30,21 @@ const FinishModal = () => {
...
-        </Button>
+        <ButtonFilled handleClick={handleClose} data-cy="selectFinish" isPrimary>
+          <div>Back to workflow</div>
+        </ButtonFilled>
       </div>
     </div>
   );
 
   return (
     <div>
```
Comment (S-ayanide) [inline litmus-portal/frontend/src/components/FinishModal/index.tsx:41]: Could have used fragments here instead

**Comment by oumkale** [comment]
Hello @rajdas98 please PTAL, you may merge this PR.

File: litmus-portal/frontend/src/components/FinishModal/styles.ts [hunk 4]
```diff
@@ -2,44 +2,37 @@ import { makeStyles, Theme } from '@material-ui/core/styles';
...
     textalign: 'center',
-    marginTop: '2rem',
+    marginTop: theme.spacing(5),
     color: theme.palette.common.black,
   },
   headWorkflow: {
-    fontfamily: 'Ubuntu',
-    fontstyle: 'normal',
-    fontweight: 'normal',
     fontsize: '16px',
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/FinishModal/styles.ts:25]: Can we change fontSize to "rem"?

File: litmus-portal/frontend/src/components/FinishModal/styles.ts [hunk 5]
```diff
@@ -2,44 +2,37 @@ import { makeStyles, Theme } from '@material-ui/core/styles';
...
   mark: {
-    marginTop: 80,
+    marginTop: theme.spacing(14),
     textAlign: 'center',
   },
   heading: {
-    fontfamily: 'Ubuntu',
-    fontstyle: 'normal',
-    fontweight: 'normal',
     fontSize: '40px',
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/FinishModal/styles.ts:19]: here also

File: litmus-portal/frontend/src/components/FinishModal/index.tsx [hunk 6]
```diff
@@ -2,31 +2,24 @@ import Button from '@material-ui/core/Button';
...
     setOpen(false);
   };
-
-  /* Body part for modal */
+  
   const body = (
     <div className={classes.rootContainer}>
-      <Icon />
+      <img src="icons/finish.png" className={classes.mark} alt="mark" />;
       <div className={classes.heading}>
```
Comment (Saranya-jena) [inline litmus-portal/frontend/src/components/FinishModal/index.tsx:23]: .svg is preferred

File: litmus-portal/frontend/src/components/FinishModal/styles.ts (hunk 4 above)
Comment (oumkale) [reply litmus-portal/frontend/src/components/FinishModal/styles.ts:25]: yes

File: litmus-portal/frontend/src/components/FinishModal/index.tsx (hunk 6 above)
Comment (oumkale) [reply litmus-portal/frontend/src/components/FinishModal/index.tsx:23]: Thank you @Saranya-jena I will look into this.

**Comment by oumkale** [comment]
Now it's good? PTAL @Saranya-jena @Jonsy13

**Comment by ishangupta-ds** [comment]
LGTM.

**Comment by oumkale** [comment]
Hello @rajdas98, PTAL

**Comment by rajdas98** [comment]
@oumkale You have to debase this PR branch. I have change the base branch from `litmus-portal` to mastee

**Comment by oumkale** [comment]
@rajdas98  PTAL. You may merge this.

**Review by ishangupta-ds — DISMISSED** [review]

**Comment by oumkale** [comment]
PTAL @Jonsy13 @arkajyotiMukherjee  @ishangupta-ds @Saranya-jena @S-ayanide

File: litmus-portal/frontend/src/components/WorkflowStepper/index.tsx [hunk 7]
```diff
@@ -130,11 +131,13 @@ const CustomStepper = () => {
     (state: RootState) => state.userData.selectedProjectID
   );
   const workflow = useActions(WorkflowActions);
+  const [validYaml, setValidYaml] = React.useState(false);
   const steps = getSteps();
 
   const handleNext = () => {
     setActiveStep((prevActiveStep) => prevActiveStep + 1);
     if (activeStep === 2) {
+      setValidYaml(false);
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/WorkflowStepper/index.tsx:140]: I think it is not required. @oumkale

File: litmus-portal/frontend/src/components/WorkflowStepper/index.tsx [hunk 8]
```diff
@@ -152,14 +155,25 @@ const CustomStepper = () => {
         weights: arr,
       });
     }
+    setValidYaml(false);
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/WorkflowStepper/index.tsx:158]: First Check, If weight.length or arr.length is 0 or not.
If 0, then don't move forward and show error.
If not 0, then move forward and don't show error.

File: litmus-portal/frontend/src/components/WorkflowStepper/index.tsx [hunk 9]
```diff
@@ -152,14 +155,25 @@ const CustomStepper = () => {
...
 
   const handleBack = () => {
+    if (activeStep !== 3) {
+      setValidYaml(false);
+    }
     setActiveStep((prevActiveStep) => prevActiveStep - 1);
   };
 
+  if (activeStep === 3) {
+    if (weights.length === 0) {
```
Comment (Jonsy13) [inline litmus-portal/frontend/src/components/WorkflowStepper/index.tsx:171]: If you check before moving forward, this won't be required.

**Review by Jonsy13 — APPROVED** [review]

**Comment by oumkale** [comment]
@arkajyotiMukherjee @S-ayanide @ishangupta-ds @amityt  @Saranya-jena  PTAL.

**Comment by oumkale** [comment]
@rajdas98 PTAL

**Review by ishangupta-ds — APPROVED** [review]
LGTM.

**Review by arkajyotiMukherjee — APPROVED** [review]
LGTM

**Pull request merged**